lib/spack/spack/variant.py

```diff
@@ -347,7 +347,11 @@ def constrain(self, other):
 
         old_value = self.value
 
-        values = list(sorted(set(self.value + other.value)))
+        if len(other.value) == 1 and other.value[0] == 'none':
+            values = ("none",)
+        else:
+            values = list(sorted(set(self.value + other.value)))
```

Comment:
I don't think this should be solved like this, since it appears to be another case of greedy concretization done by Spack original concretizer. What we are doing in the conditional looks more like overriding a spec rather than constraining it.

Reply:
Of course it's kind of bad that we don't have unit tests failing because of this.